orchestrator/test_runner/src/airdrop_proposal.rs

```diff
@@ -13,112 +15,74 @@ use deep_space::error::CosmosGrpcError;
-    submit_and_fail_airdrop_proposal(
-        bad_airdrop_denom.clone(),
-        &amounts,
-        &user_addresses,
-        contact,
-        &keys,
-        false,
-    )
-    .await;
+    submit_and_fail_airdrop_proposal(bad_airdrop_denom.clone(), &keys, contact, false, false).await;
 
     // submit an airdrop token with an invalid address
-    submit_and_fail_airdrop_proposal(
-        STAKING_TOKEN.clone(),
-        &amounts,
-        &user_addresses,
-        contact,
-        &keys,
-        true,
-    )
-    .await;
+    submit_and_fail_airdrop_proposal(STAKING_TOKEN.clone(), &keys, contact, true, false).await;
 
-    // submit the actual valid airdrop
+    // submit the actual valid airdrop, max number of possible recipients
     submit_and_pass_airdrop_proposal(
         STAKING_TOKEN.clone(),
-        &amounts,
-        &user_addresses,
-        contact,
+        NUM_AIRDROP_RECIPIENTS,
         &keys,
+        contact,
     )
     .await
     .unwrap();
 
-    wait_for_proposals_to_execute(contact).await;
-
-    // make sure everyone got their airdrop amount
-    for (key, amount) in user_addresses.iter().zip(amounts.iter()) {
-        let balances = contact
-            .get_balance(*key, STAKING_TOKEN.to_string())
+    // submit five more airdrops each with a random number of recipients
+    let mut rng = rand::thread_rng();
+    for _ in 0..3 {
+        let recipients: u16 = rng.gen_range(1..NUM_AIRDROP_RECIPIENTS);
+        submit_and_pass_airdrop_proposal(STAKING_TOKEN.clone(), recipients, &keys, contact)
             .await
             .unwrap();
-
-        assert!(balances.is_some());
-        let big_amount: Uint256 = (*amount).into();
-        assert_eq!(balances.unwrap().amount, big_amount);
     }
 
-    // try to submit the airdrop again, make sure nothing happens because we are out of tokens
-    submit_and_fail_airdrop_proposal(
-        STAKING_TOKEN.clone(),
-        &amounts,
-        &user_addresses,
-        contact,
-        &keys,
-        false,
-    )
-    .await;
-
-    let community_pool_contents_end = contact.query_community_pool().await.unwrap();
-    let end = get_coins(&*STAKING_TOKEN, &community_pool_contents_end).unwrap();
-    info!(
-        "FeePool start {} and End {}",
-        starting_amount_in_pool.amount, end.amount
-    );
-    // check that ending amount is smaller than starting (will panic on underflow)
-    // and that we have subtracted at least enough to fund the airdrop, the problem is
-    // that tokens are added to the pool via inflation while this whole test is running
-    // meaning we can't just check that it all adds up (we do that in the go unit test though)
-    assert!(starting_amount_in_pool.amount - end.amount >= 0u8.into());
-
-    info!("Successfully Issued Airdrop!");
+    // submit an airdrop that contains more tokens than are in the community pool
+    submit_and_fail_airdrop_proposal(STAKING_TOKEN.clone(), &keys, contact, false, true).await;
```

Comment:
Does it make sense to add `submit_and_fail_airdrop_proposal(...., true true)`?

Reply:
I think that would just be submit and pass airdrop proposal, which I guess means I could combine the two.